cmd/ctr/run.go

```diff
@@ -94,82 +99,131 @@ var runCommand = cli.Command{
 		if err != nil {
 			return err
 		}
-
 		imageStore, err := getImageStore(context)
 		if err != nil {
 			return errors.Wrap(err, "failed resolving image store")
 		}
-
-		if runtime.GOOS != "windows" && context.String("rootfs") == "" {
-			ref := context.Args().First()
-
-			image, err := imageStore.Get(ctx, ref)
+		differ, err := getDiffService(context)
+		if err != nil {
+			return err
+		}
+		var (
+			checkpoint      digest.Digest
+			checkpointIndex = digest.Digest(context.String("checkpoint"))
```

Comment:
use `digest.Parse` so any validation errors can be returned. We normally only cast directly to `digest.Digest` if we know the string is already validated.